nixos/modules/services/networking/frp.nix

```diff
@@ -0,0 +1,99 @@
+          The frp consists of `client` and `server`. The server is usually
+          deployed on the machine with a public IP address, and
+          the client is usually deployed on the machine
+          where the Intranet service to be penetrated resides.
+        '';
+      };
+
+      settings = mkOption {
+        type = settingsFormat.type;
+        default = { };
+        description = lib.mdDoc ''
+          Frp configuration, for configuration options
+          see the example of [client](https://github.com/fatedier/frp/blob/dev/conf/frpc_full.ini)
+          or [server](https://github.com/fatedier/frp/blob/dev/conf/frps_full.ini) on github.
+        '';
+        example = literalExpression ''
+          {
+            common = {
+              server_addr = "x.x.x.x";
+              server_port = 7000;
+            };
+          }
+        '';
+      };
+    };
+  };
+
+  config =
+    let
+      serviceWants = lib.optionals isClient [ "network-online.target" ];
+      serviceAfter = if isClient then [ "network-online.target" ] else [ "network.target" ];
+      serviceRestrictAddressFamilies = [ "AF_INET" "AF_INET6" ] ++ lib.optionals isClient [ "AF_UNIX" ];
+      serviceCapability = lib.optionals isServer [ "CAP_NET_BIND_SERVICE" ];
+      serviceStateDirectoryMode = lib.optionalString isServer "0700";
+      serviceUMask = lib.optionalString isServer "0007";
+      executableFile = if isClient then "frpc" else "frps";
+    in
+    lib.mkIf cfg.enable {
+      systemd.services = {
+        frp = {
+          wants = serviceWants;
+          after = serviceAfter;
+          wantedBy = [ "multi-user.target" ];
+          description = "A fast reverse proxy to help you expose a local server behind a NAT or firewall to the internet.";
+          serviceConfig = {
+            Type = "simple";
+            Restart = "on-failure";
+            RestartSec = 15;
+            ExecStart = "${cfg.package}/bin/${executableFile} -c ${configFile}";
+            StateDirectoryMode = serviceStateDirectoryMode;
+            UMask = serviceUMask;
+            # Hardening
+            DynamicUser = true;
+            CapabilityBoundingSet = serviceCapability;
+            AmbientCapabilities = serviceCapability;
+            PrivateDevices = true;
+            ProtectHostname = true;
+            ProtectClock = true;
+            ProtectKernelTunables = true;
+            ProtectKernelModules = true;
+            ProtectKernelLogs = true;
+            ProtectControlGroups = true;
+            RestrictAddressFamilies = serviceRestrictAddressFamilies;
+            LockPersonality = true;
+            MemoryDenyWriteExecute = true;
+            RestrictRealtime = true;
+            RestrictSUIDSGID = true;
+            PrivateMounts = true;
+            SystemCallArchitectures = "native";
+            SystemCallFilter = [ "@system-service" ];
+          };
+        };
+      };
+    };
+
+  meta.maintainers = with maintainers; [ zaldnoay ];
```

Comment:
@Janik-Haag Do you want to be a maintainer?

Reply:
Thank you for asking but since I didn't use the tool in the last few month I would probably say no.